python/pyspark/mllib/linalg.py

```diff
@@ -1152,9 +1156,385 @@ def sparse(numRows, numCols, colPtrs, rowIndices, values):
         return SparseMatrix(numRows, numCols, colPtrs, rowIndices, values)
 
 
+class DistributedMatrix(object):
+    """Represents a distributively stored matrix backed by one or more RDDs."""
+    def numRows(self):
+        """Get or compute the number of rows."""
+        raise NotImplementedError
+
+    def numCols(self):
+        """Get or compute the number of cols."""
+        raise NotImplementedError
+
+
+class DistributedMatrices(object):
+    """Factory methods for distributed matrices."""
+    @staticmethod
+    def rowMatrix(rows, numRows=0, numCols=0):
+        """
+        Create a RowMatrix.
+
+        :param rows: An RDD of Vectors.
+        """
+        javaRowMatrix = callMLlibFunc("createRowMatrix", rows, long(numRows), int(numCols))
+        jrm = JavaModelWrapper(javaRowMatrix)
+        return RowMatrix(jrm)
+
+    @staticmethod
+    def indexedRowMatrix(rows, numRows=0, numCols=0):
+        """
+        Create an IndexedRowMatrix.
+
+        :param rows: An RDD of IndexedRows or (long, Vector) tuples.
+        """
+        # We use DataFrames for serialization of IndexedRows from Python, so convert the RDD to a
+        # DataFrame. This will convert each IndexedRow to a Row containing the 'index' and 'vector'
+        # values, which can both be easily serialized.  We will convert back to IndexedRows on the
+        # Scala side.
+        javaIndexedRowMatrix = callMLlibFunc("createIndexedRowMatrix", rows.toDF(),
+                                             long(numRows), int(numCols))
+        jirm = JavaModelWrapper(javaIndexedRowMatrix)
+        return IndexedRowMatrix(jirm)
+
+    @staticmethod
+    def coordinateMatrix(entries, numRows=0, numCols=0):
+        """
+        Create a CoordinateMatrix.
+
+        :param entries: An RDD of MatrixEntry inputs or (long, long, float) tuples.
+        """
+        # We use DataFrames for serialization of MatrixEntry inputs from Python, so convert the RDD
+        # to a DataFrame. This will convert each MatrixEntry to a Row containing the 'i', 'j', and
+        # 'value' values, which can each be easily serialized.  We will convert back to MatrixEntry
+        # inputs on the Scala side.
+        javaCoordinateMatrix = callMLlibFunc("createCoordinateMatrix", entries.toDF(),
+                                             long(numRows), long(numCols))
+        jcm = JavaModelWrapper(javaCoordinateMatrix)
+        return CoordinateMatrix(jcm)
+
+
+class RowMatrix(DistributedMatrix):
+    """
+    Represents a row-oriented distributed Matrix with no meaningful row indices.
+
+    .. note:: Experimental
+    """
+    def __init__(self, jrm):
+        """ Create a wrapper over a Java RowMatrix. """
+        self._jrm = jrm
+        self.rows = self._rows()
+
+    def _rows(self):
+        """
+        Get the rows of the RowMatrix as a RDD of Vectors.
+
+        >>> rows = sc.parallelize([Vectors.dense([1, 2, 3]), Vectors.dense([4, 5, 6]),
```

Comment:
There are some rules in PEP8 like if we limit the line length to 100 in code, we need to limit the docstring length to 72. I don't understand why, but since they are rules it might be better to follow them :) (http://legacy.python.org/dev/peps/pep-0008/#maximum-line-length)